Frontend/ui-customer/src/Screens/CustomerUtils/RideSelectionScreen/Controller.purs

```diff
@@ -0,0 +1,276 @@
+    RideBookingListAPIResponseAction rideList status -> trackAppScreenEvent appId (getScreen RIDE_SELECTION_SCREEN) "in_screen" "ride_booking_list"
+    NoAction -> trackAppScreenEvent appId (getScreen RIDE_SELECTION_SCREEN) "in_screen" "no_action"
+    _     -> pure unit
+
+data ScreenOutput = GoBack
+                  | SelectRide    RideSelectionScreenState
+                  | LoaderOutput  RideSelectionScreenState
+                  | RefreshScreen RideSelectionScreenState
+
+data Action = NoAction
+            | OnFadeComplete String
+            | Refresh
+            | Loader
+            | BackPressed
+            | GenericHeaderActionController GenericHeader.Action
+            | RideBookingListAPIResponseAction RideBookingListRes String
+            | IndividualRideCardActionController IndividualRideCardController.Action
+            | APIFailureActionController ErrorModal.Action
+            | ErrorModalActionController   ErrorModal.Action
+            | Scroll String
+            | AfterRender
+            | ScrollStateChanged ScrollState
+            | DontKnowRide       PrimaryButton.Action
+
+
+eval :: Action -> RideSelectionScreenState -> Eval Action ScreenOutput RideSelectionScreenState
+
+eval (IndividualRideCardActionController (IndividualRideCardController.OnClick index)) state = do
+  let selectedCard = state.itemsRides !! index
+  case selectedCard of
+    Just selectedRide -> do
+      exit $ SelectRide state  { selectedItem = Just selectedRide}
+    Nothing -> continue state
+
+eval (DontKnowRide (PrimaryButton.OnClick)) state = do
+  exit $ SelectRide state {
+      selectedItem = Nothing
+  }
+
+eval BackPressed state = exit GoBack
+
+eval (ScrollStateChanged scrollState) state = do
+  _ <- case scrollState of
+        SCROLL_STATE_FLING -> pure $ setEnabled "2000031" false
+        _ -> pure unit
+  continue state
+
+eval (GenericHeaderActionController (GenericHeader.PrefixImgOnClick )) state = continueWithCmd state [do pure BackPressed]
+
+eval (OnFadeComplete _ ) state = do
+  if (state.props.receivedResponse == false) then continue state
+  else  continue state {
+          shimmerLoader = case state.shimmerLoader of
+            AnimatedIn ->AnimatedOut
+            AnimatingOut -> AnimatedOut
+            a -> a
+        }
+
+eval (Loader) state = updateAndExit state{shimmerLoader = AnimatedIn, props{loaderButtonVisibility = false}} $ LoaderOutput state
+
+eval (Scroll value) state = do
+  let firstIndex = fromMaybe 0 (fromString (fromMaybe "0"((split (Pattern ",")(value))!!0)))
+  let visibleItems = fromMaybe 0 (fromString (fromMaybe "0"((split (Pattern ",")(value))!!1)))
+  let totalItems = fromMaybe 0 (fromString (fromMaybe "0"((split (Pattern ",")(value))!!2)))
+  let canScrollUp = fromMaybe true (strToBool (fromMaybe "true" ((split (Pattern ",")(value))!!3)))
+  let loadMoreButton = if (totalItems == (firstIndex + visibleItems) && totalItems /= 0 && totalItems /= visibleItems) then true else false
```

Comment:
`(totalItems == (firstIndex + visibleItems) && totalItems /= 0 && totalItems /= visibleItems)` can be assigned directly here. no need of `then true else false`